generate/generate.go

```diff
@@ -0,0 +1,1015 @@
+	g.spec.Hooks.Prestart = append(g.spec.Hooks.Prestart, hook)
+	return nil
+}
+
+// ClearPostStopHooks clear g.spec.Hooks.Poststop.
+func (g Generator) ClearPostStopHooks() {
+	g.spec.Hooks.Poststop = []rspec.Hook{}
+}
+
+// AddPostStopHook adds a poststop hook into g.spec.Hooks.Poststop.
+func (g Generator) AddPostStopHook(s string) error {
+	hook := parseHook(s)
+	g.spec.Hooks.Poststop = append(g.spec.Hooks.Poststop, hook)
+	return nil
+}
+
+// ClearPostStartHooks clear g.spec.Hooks.Poststart.
+func (g Generator) ClearPostStartHooks() {
+	g.spec.Hooks.Poststart = []rspec.Hook{}
+}
+
+// AddPostStartHook adds a poststart hook into g.spec.Hooks.Poststart.
+func (g Generator) AddPostStartHook(s string) error {
+	hook := parseHook(s)
+	g.spec.Hooks.Poststart = append(g.spec.Hooks.Poststart, hook)
+	return nil
+}
+
+// AddTmpfsMount adds a tmpfs mount into g.spec.Mounts.
+func (g Generator) AddTmpfsMount(dest string) error {
+	mnt := rspec.Mount{
+		Destination: dest,
+		Type:        "tmpfs",
+		Source:      "tmpfs",
+		Options:     []string{"nosuid", "nodev", "mode=755"},
+	}
+
+	g.spec.Mounts = append(g.spec.Mounts, mnt)
+	return nil
+}
+
+// AddCgroupsMount adds a cgroup mount into g.spec.Mounts.
+func (g Generator) AddCgroupsMount(mountCgroupOption string) error {
+	switch mountCgroupOption {
+	case "ro":
+	case "rw":
+	case "no":
+		return nil
+	default:
+		return fmt.Errorf("--mount-cgroups should be one of (ro,rw,no)")
+	}
+
+	mnt := rspec.Mount{
+		Destination: "/sys/fs/cgroup",
+		Type:        "cgroup",
+		Source:      "cgroup",
+		Options:     []string{"nosuid", "noexec", "nodev", "relatime", mountCgroupOption},
+	}
+	g.spec.Mounts = append(g.spec.Mounts, mnt)
+
+	return nil
+}
+
+// AddBindMount adds a bind mount into g.spec.Mounts.
+func (g Generator) AddBindMount(bind string) error {
+	var source, dest string
+	options := "ro"
+	bparts := strings.SplitN(bind, ":", 3)
+	switch len(bparts) {
+	case 2:
+		source, dest = bparts[0], bparts[1]
+	case 3:
+		source, dest, options = bparts[0], bparts[1], bparts[2]
+	default:
+		return fmt.Errorf("--bind should have format src:dest:[options]")
+	}
+
+	defaultOptions := []string{"bind"}
+	mnt := rspec.Mount{
+		Destination: dest,
+		Type:        "bind",
+		Source:      source,
+		Options:     append(defaultOptions, options),
+	}
+	g.spec.Mounts = append(g.spec.Mounts, mnt)
+	return nil
+}
+
+// SetupPrivileged sets up the priviledge-related fields inside g.spec.
+func (g Generator) SetupPrivileged(privileged bool) {
+	if privileged {
+		// Add all capabilities in privileged mode.
+		var finalCapList []string
+		for _, cap := range capability.List() {
+			finalCapList = append(finalCapList, fmt.Sprintf("CAP_%s", strings.ToUpper(cap.String())))
+		}
+		g.spec.Process.Capabilities = finalCapList
+		g.spec.Process.SelinuxLabel = ""
+		g.spec.Process.ApparmorProfile = ""
+		g.spec.Linux.Seccomp = nil
+	}
+}
+
+func checkCap(c string) error {
+	isValid := false
+	cp := fmt.Sprintf("CAP_%s", c)
```

Comment:
The list I am checking against is `capability.List()`, which includes `CAP_CHOWN`, `CAP_DAC_OVERRIDE`, and so on.